concert/imageprocessing.py

```diff
@@ -78,6 +89,81 @@ def find_needle_tip(image):
 
     return np.mean(coords, axis=0) if coords else None
 
+def get_sphere_absorption_pattern(radius: int) -> ArrayLike:
+    """
+    Returns a generic spherical absorption pattern
+    :param radius: radius for simulating pattern
+    :type radius: int
+    :return: absorption pattern
+    :rtype: ArrayLike
+    """
+    y, x = xp.mgrid[-radius:radius + 1, -radius:radius + 1]
+    mask: ArrayLike = xp.where(radius ** 2 - x ** 2 - y ** 2 >= 0)
+    sphere: ArrayLike = xp.zeros((2 * radius + 1, 2 * radius + 1))
+    sphere[mask] = 2 * xp.sqrt(radius ** 2 - x[mask] ** 2 - y[mask] ** 2)
+    return sphere
+
+
+def get_sphere_center_corr(proj: ArrayLike, sphere: ArrayLike,
+                           radius: int, **kwargs) -> Tuple[int, int]:
```

Comment:
If the sphere is perfect you don't need a reference simulation, right? You can use the previous image and cross-correlate that. That would have the big advantage that people wouldn't need to take care about the radius. We should discuss this.